modules/imgcodecs/test/test_animation.cpp

```diff
@@ -638,4 +638,42 @@ TEST(Imgcodecs_APNG, imencode_animation)
 
 #endif // HAVE_PNG
 
+#if defined(HAVE_PNG) || defined(HAVE_SPNG)
+
+TEST(Imgcodecs_APNG, imread_animation_16u)
+{
+    // Set the path to the test image directory and filename for loading.
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filename = root + "readwrite/033.png";
+
+    Mat img = imread(filename, IMREAD_UNCHANGED);
+    EXPECT_FALSE(img.empty());
+    EXPECT_TRUE(img.type() == CV_16UC4);
+    EXPECT_EQ(0,     img.at<ushort>(0, 0));
+    EXPECT_EQ(0,     img.at<ushort>(0, 1));
+    EXPECT_EQ(65280, img.at<ushort>(0, 2));
+    EXPECT_EQ(65535, img.at<ushort>(0, 3));
+
+    img = imread(filename, IMREAD_GRAYSCALE);
+    EXPECT_FALSE(img.empty());
+    EXPECT_TRUE(img.type() == CV_8UC1);
```

Comment:
Is 033.png 16 bit or 8 bit? Asking for grayscale should keep the depth

Reply:
033.png is `CV_16UC4`

IMREAD_GRAYSCALE converts it to `CV_8UC1` by default.

let me check `IMREAD_GRAYSCALE | IMREAD_ANYDEPTH`